File: code-examples/php/usage-examples/updateOne.php
<?php

require 'vendor/autoload.php';

// Replace the uri string with your MongoDB deployment's connection string
$uri = "mongodb://localhost:27017";

$client = new MongoDB\Client($uri);

// Get the "movies" collection in the "sample_mflix" database
$collection = $client->sample_mflix->movies;

try {
    /* Update a document that has the title "Random Harvest" to have a
    plot field with the specified value */
    $result = $collection->updateOne(
        [ 'title' => 'Random Harvest' ],
        [ '$set' => [
               'plot' => "A harvest of random numbers, such as: " . rand(),
            ],
        ],
       // Set the upsert option to insert a document if no documents
       // match the filter
        [ 'upsert' => true ]
    );

    // Print the number of matched and modified documents
    printf("%d document(s) matched the filter, updated %d document(s)\n",
        $result->getMatchedCount(), $result->getModifiedCount()
    );

} catch (Exception $e) {
    echo $e->getMessage();
}

?>

File: code-examples/scala/usage-examples/updateOne.scala
import org.mongodb.scala._
import scala.util.Random

// Replace the uri string with your MongoDB deployment's connection string
val mongoClient: MongoClient = MongoClient("<connection string uri>")

val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
val collection: MongoCollection[Document] = database.getCollection("movies")

case class Movie(plot: String, title: String)

val plotString = s"A harvest of random numbers, such as: ${Random.nextFloat()}"

val filter: Bson = Filters.eq("title", "Random Harvest")
val update: Bson = Updates.set("plot", plotString)

collection.updateOne(filter, update, new UpdateOptions().upsert(true)).subscribe(
  (updateResult: UpdateResult) => {
    println(s"${updateResult.getMatchedCount} document(s) matched the filter, updated ${updateResult.getModifiedCount} document(s)")
    mongoClient.close()
  },
  (t: Throwable) => t.printStackTrace()
)
